projects/material-addons/src/lib/quick-list/reactive-form-quick-list/reactive-form-quick-list.component.cy.ts

```diff
@@ -0,0 +1,234 @@
+                <input formControlName="name" matInput />
+              </mat-form-field>
+            </mad-readonly-form-field-wrapper>
+          </span>
+        </form>
+      </ng-template>
+    </mad-reactive-form-quick-list>
+  </form>`,
+  styles: `
+    mad-readonly-form-field-wrapper:not(:first-child) {
+      margin-left: 2em;
+    }
+  `,
+})
+class TestWrapperComponent implements OnInit {
+  addLabel = 'Add';
+  minItems = 0;
+  maxItems = 3;
+  formGroup: FormGroup;
+  textIsEditable = false;
+  addPossible = true;
+  removePossible = true;
+
+  constructor(
+    private formBuilder: FormBuilder,
+    private cd: ChangeDetectorRef,
+  ) {}
+
+  get array(): FormArray {
+    return this.formGroup?.get('array') as FormArray;
+  }
+
+  ngOnInit(): void {
+    this.formGroup = this.formBuilder.group({
+      test: 'test',
+      array: this.formBuilder.array([
+        this.formBuilder.group({
+          id: 1,
+          name: 'Name 1',
+        }),
+        this.formBuilder.group({
+          id: 2,
+          name: 'Name 2',
+        }),
+      ]),
+    });
+  }
+
+  onItemAdded(): void {
+    this.array.push(
+      this.formBuilder.group({
+        id: null,
+        name: null,
+      }),
+    );
+    this.formGroup.updateValueAndValidity();
+    this.cd.detectChanges(); // to avoid ExpressionChangedAfterItHasBeenCheckedError on add button click
+  }
+
+  onItemRemoved(): void {}
+}
+
+describe('reactive-form-quick-list.component.cy.ts', () => {
+  beforeEach(() => {
+    cy.viewport(600, 600);
+  });
+
+  it('should mount ReactiveFormQuickListComponent', () => {
+    mountWrapperComponent();
+  });
+
+  it('should not show add and delete buttons in readonly mode', () => {
+    mountWrapperComponent();
+
+    cy.getByCySel('reactive-form-quick-list').should('be.visible').and('have.attr', 'ng-reflect-readonly', 'true');
+    cy.getByCySel('reactive-form-quick-list-item').should('be.visible');
+    cy.getByCySel('add-item-button').should('not.exist');
+    cy.getByCySel('delete-item-button').should('not.exist');
+  });
+
+  it('should mount ReactiveFormQuickListComponent', () => {
+    mountWrapperComponent({
+      textIsEditable: true,
+    });
+
+    cy.getByCySel('reactive-form-quick-list').should('be.visible');
+    cy.getByCySel('reactive-form-quick-list-item').should('be.visible');
+    cy.getByCySel('add-item-button').should('be.visible');
+    cy.getByCySel('delete-item-button').should('be.visible');
+  });
+
+  it('should display proper add button label', () => {
+    mountWrapperComponent({
+      textIsEditable: true,
+      addLabel: 'Test Label',
+    });
+    cy.getByCySel('add-item-button').should('contain.text', ' Test Label\n');
+  });
+
+  it('should display the correct number of items', () => {
+    mountWrapperComponent();
+    cy.getByCySel('reactive-form-quick-list-item').should('have.length', 2);
+  });
+
+  it('should add an item when add button is clicked', () => {
```

Comment:
Aren't there very similar things inside the quick-list component tests?

Reply:
Fixed with stepper.component.cy.ts tests